pkg/controller/ipam/antrea_ipam_controller.go

```diff
@@ -187,40 +212,119 @@ func (c *AntreaIPAMController) cleanIPPoolForStatefulSet(namespacedName string)
 	return nil
 }
 
+// Find IP Pools annotated to StatefulSet via direct annotation or namespace annotation
+func (c *AntreaIPAMController) getIPPoolsForStatefulSet(ss *appsv1.StatefulSet) []string {
+
+	// Inspect pool annotation for the Pods
+	// In order to avoid extra API call in IPAM driver, IPAM annotations are defined
+	// on Pods rather than on StatefulSet
+	annotations, exists := ss.Spec.Template.Annotations[annotation.AntreaIPAMAnnotationKey]
+	if exists {
+		// Stateful Set Pod is annotated with dedicated IP pool
+		return strings.Split(annotations, annotation.AntreaIPAMAnnotationDelimiter)
+	}
+
+	// Inspect Namespace
+	namespace, err := c.namespaceLister.Get(ss.Namespace)
+	if err != nil {
+		// Should never happen
+		klog.Errorf("Namespace %s not found for StatefulSet %s", ss.Namespace, ss.Name)
```

Comment:
Can we assume the error is "namespace not found"?

Reply:
Sorry, not sure what you mean, could you please elaborate? This error is returned when `Get` fails.

Reply:
Thanks @gran-vmv !

Reply:
In general, we should klog.ErrorS().